device_tracker.py

```
@@ -47,9 +50,20 @@ async def async_setup_entry(
     """Set up device tracker for AT&T Router."""
     session = aiohttp_client.async_get_clientsession(hass)
-    client = ATTRouterClient(
-        session, entry.data[CONF_HOST], entry.data[CONF_SESSION_ID]
-    )
+    
+    # Create client based on auth method
+    if entry.data.get(CONF_AUTH_METHOD) == AUTH_METHOD_ACCESS_CODE:
+        client = ATTRouterClient(
+            session,
+            entry.data[CONF_HOST],
+            access_code=entry.data.get(CONF_ACCESS_CODE)
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)